src/openfermion/measurements/_bayesian_estimators.py

```diff
@@ -0,0 +1,520 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""
+Classes for Bayesian estimators for quantum phase estimation.
+Estimators described in arXiv:1809.09697 (in particular in App.C and App.C.1)
+"""
+import copy
+import warnings
+import numpy
+from numpy import cos, sin, pi
+from scipy import optimize
+from openfermion.config import EQ_TOLERANCE
+from ._probability_distributions import (
+    FourierProbabilityDist,
+    make_fourier_matrices,
+    make_fourier_matrix)
+
+# TODO - shift p_vec functions to probability_distributions,
+# check if tests pass.
+
+class BayesEstimator(object):
+    """
+    Class to estimate a QPE experiment in the absence of error.
+    """
+
+    def __init__(self,
+                 prior_amplitude_mean=None,
+                 prior_amplitude_var=None,
+                 num_vectors=1,
+                 num_freqs=1000,
+                 max_rotations=1,
+                 prior_vector=None,
```

Comment:
Instead of taking the arguments `prior_amplitude_mean`, `prior_amplitude_var`, `num_vectors`, `num_freqs`, and `prior_vector`, just have one argument `prior` which is a FourierProbabilityDist.

Reply:
Good idea, I'll implement this now.